crates/bdk/examples/compiler.rs

```diff
@@ -42,12 +42,23 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     // Create a `wsh` type descriptor from the policy.
     // `policy.compile()` returns the resulting miniscript from the policy.
-    let descriptor = Descriptor::new_wsh(policy.compile()?)?;
+    let descriptor = Descriptor::new_wsh(policy.compile()?)?.to_string();
 
-    println!("Compiled into following Descriptor: \n{}", descriptor);
+    println!("Compiled into Descriptor: \n{}", descriptor);
 
-    // Create a new wallet from this descriptor
-    let mut wallet = Wallet::new_no_persist(&format!("{}", descriptor), None, Network::Regtest)?;
+    // Do the same for another (internal) keychain
+    let policy_str = "or(10@thresh(2,pk(029ffbe722b147f3035c87cb1c60b9a5947dd49c774cc31e94773478711a929ac0),pk(025f05815e3a1a8a83bfbb03ce016c9a2ee31066b98f567f6227df1d76ec4bd143),pk(025625f41e4a065efc06d5019cbbd56fe8c07595af1231e7cbc03fafb87ebb71ec)),1@and(pk(03deae92101c790b12653231439f27b8897264125ecb2f46f48278603102573165),older(12960)))";
+    println!("Compiling internal policy: \n{}", policy_str);
+
+    let policy = Concrete::<String>::from_str(policy_str)?;
+    let internal_descriptor = Descriptor::new_wsh(policy.compile()?)?.to_string();
+    println!(
+        "Compiled into internal Descriptor: \n{}",
+        internal_descriptor
+    );
```

Comment:
We are using edition 2021, so you can do:
```suggestion
    println!("Compiled into internal Descriptor: \n{internal_descriptor}");
```

Reply:
https://github.com/bitcoindevkit/bdk/pull/1390#discussion_r1545444089